addons/crm_survey/models/survey_user_input.py

```diff
@@ -0,0 +1,180 @@
+            answer_description, last_row_id = "", ""
+            # Since if no response is given by the user, an input is still made with the "skipped" field set to True,
+            # that's why we retrieve the first "response" of user inputs to test it.
+            if not input_lines[0].skipped:
+                for answer_index, answer_id in enumerate(input_lines):
+                    # Check if the question has a nickname recorded or an email answer
+                    if question_id.question_type == "char_box":
+                        if not user_nickname and question_id.save_as_nickname:
+                            user_nickname = answer_id.display_name
+                        if not public_user_mail and question_id.validation_email:
+                            public_user_mail = answer_id.display_name
+
+                    # Write description line
+                    if question_id.question_type == "text_box":
+                        answer_description = Markup("<br/>&emsp;%s" % escape(answer_id._get_answer_value()))
+                    elif question_id.question_type == "matrix":
+                        row_id = answer_id.matrix_row_id
+                        if row_id.display_name:
+                            col_value = answer_id.suggested_answer_id.display_name
+                            if last_row_id != row_id:
+                                last_row_id = row_id
+                                answer_description += Markup("<br/>&emsp;%s — %s" % (escape(row_id.display_name), escape(col_value)))
+                            else:  # For multiple suggested answers
+                                answer_description += Markup(", %s" % escape(col_value))
+                        else:  # Comment field
+                            comment_answer = answer_id._get_answer_value()
+                            answer_description += Markup("<br/>&emsp;%s — %s" % (_('Comment'), escape(comment_answer)))
+                    else:
+                        answer_description += Markup("%s" % (" — " if not answer_index else ", "))   # For multiple suggested answers
+                        if question_id.question_type in ["char_box", "simple_choice", "multiple_choice"]:  # User input escaped (comment field or char box)
+                            answer = answer_id.display_name
+                        else:  # Normal case
+                            answer = answer_id.display_name
+                        if answer_id.suggested_answer_id and answer_id.display_name == "Skipped":
+                            answer = answer_id.suggested_answer_id.value_image_filename
+                        answer_description += Markup("%s" % escape(answer))
+            else:
+                # To write "Skipped" in italic
+                answer_description = Markup("<i> — %s </i>" % _('Skipped'))
+
+            answer_description = Markup("%s" % answer_description.replace('\n', Markup("<br/>&emsp;")))  # To keep line HTML breaks
```

Comment:
Should we not do that inside the loop (for text then?), and remove this whole line ? :)